lib/hexpm_web/templates/dashboard/_keys.html.eex

```diff
@@ -55,37 +55,16 @@
         <%= ViewHelpers.text_input f, :name, class: "form-control", placeholder: "Name" %>
         <%= ViewHelpers.error_tag f, :name %>
       </div>
-      <%= inputs_for f, :permissions, fn f -> %>
-        <%= label f, :domain, "Permission domain" %>
-        <div class="form-group">
-          <%=
-            ViewHelpers.select(
-              f,
-              :domain,
-              [
-                API: "api",
-                "Organization": "repository"
-              ],
-              class: "form-control"
-            )
-          %>
-          <%= ViewHelpers.error_tag f, :domain %>
-        </div>
-        <%= label f, :domain, "Domain resource" %>
-        <div class="form-group">
-          <%=
-            ViewHelpers.select(
-              f,
-              :resource,
-              [
-                API: ["Read": "read", "Write": "write"],
-                "Organization": organization_resources(assigns),
-              ],
-              class: "form-control"
-            )
-          %>
-          <%= ViewHelpers.error_tag f, :resource %>
-        </div>
+      <span class="form-label">Key permissions</span>
+      <div class="checkbox"><label><input type="checkbox" name="key[permissions][api][read]">API: read</label></div>
+      <div class="checkbox"><label><input type="checkbox" name="key[permissions][api][write]">API: write</label></div>
+      <%= if assigns[:organization] do %>
+        <div class="checkbox"><label><input type="checkbox" name="key[permissions][repository][<%= @organization.name %>]">Organization repository</label></div>
+      <% else %>
+        <div class="checkbox"><label><input type="checkbox" name="key[permissions][repositories]">All Repositories</label></div>
```

Comment:
We could add some sprinkles that checking "All repositories" unchecks individual repository permissions and checking individuals unchecks "All repositories". This is a minor enhancements that I can work on after this lands.

Btw, this feature is 😍

Reply:
Yeah I was thinking of maybe grouping them so that you have:

```
[ ] API
  [ ] Read
  [ ] Write
[ ] Repository
  [ ] acme
  [ ] myrepo
```

Checking API or Repository also checks the subgroups and disables them. I was hoping I could do this with only CSS but it seems it is not possible. Should we use liveview or write javascript for this?

Reply:
Oh this UI is even better! Regarding tech either way is fine. Bringing in LV just for this feels like an overkill but I’m sure we could use it elsewhere, maybe it makes sense identifying other parts first that would take advantage of it. Either case, up to you!

Reply:
Your current version is a little unwieldy, but I dont think it is common for someone to have access to that many repos.
I think liveview would be fine, but I suspect we could get the job done with jQuery as well.